src/SDSS.jl

```diff
@@ -130,5 +135,454 @@ function load_stamp_catalog(cat_dir, stamp_id, blob; match_blob=false)
+    post = gmmposterior(gmm, x_mat) 
+
+    while !fit_done
+        # Update gmm using last value of post.  post[1] contains
+        # posterior probabilities of the indicators.
+        z = post[1] .* psf_mat
+        z_sum = collect(sum(z, 1))
+        new_w = z_sum / sum(z_sum)
+        gmm.w = new_w
+        for d=1:gmm.n
+            if new_w[d] > 1e-6
+                new_mean = sum(x_mat .* z[:, d], 1) / z_sum[d]
+                x_centered = broadcast(-, x_mat, new_mean)
+                x_cov = x_centered' * (x_centered .* z[:, d]) / z_sum[d]
+
+                gmm.μ[d, :] = new_mean
+                gmm.Σ[d] = sigma_for_gmm(x_cov)
+            else
+                warn("Component $d has very small probability.")
+            end
+        end
+
+        # Get next posterior and check for convergence.  post[2] contains
+        # the log densities at each point.
+        post = gmmposterior(gmm, x_mat) 
+        gmm_fit = exp(post[2]) * gmm.w;
+        err = mean((gmm_fit - psf_mat) .^ 2)
+        err_diff = last_err - err
+        last_err = err
+        if isnan(err)
+            error("NaN in MVN PSF fit.")
+        end
+
+        iter = iter + 1
+        if err_diff < tol
+            fit_done = true
+        elseif iter >= max_iter
+            warn("PSF MVN fit: max_iter exceeded")
+            fit_done = true
+        end
+
+        println("Fitting psf: $iter: $err_diff")
+    end
+
+    gmm
+end
+
+
+@doc """
+Convert a GaussianMixtures GMM object to an array of Celect PsfComponents.
+
+Args:
+ - gmm: A GaussianMixtures GMM object (e.g. as returned by fit_psf_gaussians)
+
+ Returns:
+  - An array of PsfComponent objects.
+""" ->
+function convert_gmm_to_celeste(gmm::GMM)
+    function convert_gmm_component_to_celeste(gmm::GMM, d)
+        PsfComponent(gmm.w[d], collect(means(gmm)[d, :]), covars(gmm)[d])
+    end
+
+    [ convert_gmm_component_to_celeste(gmm, d) for d=1:gmm.n ]
+end
+
+
+@doc """
+Using data from a psField file, evaluate the PSF for a source at given point.
+
+Args:
+ - row: The row of the point source (may be a float).
+ - col: The column of the point source (may be a float).
+ - rrows: An (rnrow * rncol) by (k) matrix of flattened eigenimages.
+ - rnrow: The number of rows in the eigenimage.
+ - rncol: The number of columns in the eigenimage.
+ - cmat: An (:, :, k)-array of coefficients of the weight polynomial.
+
+Returns:
+ - An rnrow x rncol image of the PSF at (row, col)
+
+The PSF is represented as a weighted combination of "eigenimages" (stored
+in rrows), where the weights vary smoothly across points (row, col) in the image
+as a polynomial of the form
+weight[k](row, col) = sum_{i,j} cmat[i, j, k] * (rcs * row) ^ i (rcs * col) ^ j
+
+This function is based on the function getPsfAtPoints in astrometry.net:
+https://github.com/dstndstn/astrometry.net/blob/master/sdss/common.py#L953
+""" ->
+function get_psf_at_point(row::Float64, col::Float64,
```

Comment:
Yeah, there's a lot of this PSF code...I think it'd be good to have it all together in it's own module.